File: ios/ReactNativeLiveStreamViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(ReactNativeLiveStreamViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(audio, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(video, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(camera, NSString)
RCT_EXPORT_VIEW_PROPERTY(isMuted, BOOL)
RCT_EXPORT_VIEW_PROPERTY(enablePinchedZoom, BOOL)
RCT_EXPORT_VIEW_PROPERTY(zoomRatio, double)

RCT_EXPORT_VIEW_PROPERTY(onConnectionSuccess, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onConnectionFailed, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onDisconnect, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onStartStreaming, RCTDirectEventBlock)

RCT_EXTERN_METHOD(
                  startStreamingFromManager:(nonnull NSNumber *)node withRequestId:(nonnull NSNumber)requestId withStreamKey:(nonnull NSString)streamKey withUrl:(NSString)url
)
RCT_EXTERN_METHOD(
    stopStreamingFromManager:(nonnull NSNumber *)node
)
RCT_EXTERN_METHOD(
                  zoomRatioFromManager:(nonnull NSNumber *)node withZoomRatio:(nonnull NSNumber)zoomRatio
)

@end

File: ios/ReactNativeLiveStreamViewManager.swift
import ApiVideoLiveStream
import CoreGraphics
import Foundation

@objc(ReactNativeLiveStreamViewManager)
class ReactNativeLiveStreamViewManager: RCTViewManager {
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    override func view() -> (ReactNativeLiveStreamView) {
        return ReactNativeLiveStreamView()
    }

    @objc func startStreamingFromManager(_ node: NSNumber, withRequestId requestId: NSNumber, withStreamKey streamKey: String, withUrl url: String?) {
        DispatchQueue.main.async {
            let component = self.bridge.uiManager.view(
                forReactTag: node
            ) as! ReactNativeLiveStreamView
            component.startStreaming(requestId: Int(truncating: requestId), streamKey: streamKey, url: url)
        }
    }

    @objc func stopStreamingFromManager(_ node: NSNumber) {
        DispatchQueue.main.async {
            let component = self.bridge.uiManager.view(
                forReactTag: node
            ) as! ReactNativeLiveStreamView
            component.stopStreaming()
        }
    }

    @objc func zoomRatioFromManager(_ node: NSNumber, withZoomRatio zoomRatio: NSNumber) {
        DispatchQueue.main.async {
            let component = self.bridge.uiManager.view(
                forReactTag: node
            ) as! ReactNativeLiveStreamView
            component.setZoomRatio(zoomRatio: CGFloat(zoomRatio))
        }
    }
}
